validate/data.go

```diff
@@ -11,11 +13,18 @@ import (
 // EntryReader consumes a reader & returns any validation errors present
 // TODO - refactor this to wrap a reader & return a struct that gives an
 // error or nil on each entry read.
-func EntryReader(r dsio.EntryReader) ([]jsonschema.ValError, error) {
+func EntryReader(r dsio.EntryReader) ([]jsonschema.KeyError, error) {
+	ctx := context.Background()
+	batchSize := 5000
```

Comment:
Please move this to a top-level const

Or, alternatively, make it a top-level var so that it can be overridden by tests, which allows easy exercising of the batch behavior.